lib/ansible/plugins/filter/core.py

```diff
@@ -189,6 +194,7 @@ def ternary(value, true_val, false_val, none_val=None):
 
 
 def regex_escape(string, re_type='python'):
+    string = to_text(string, errors='surrogate_or_strict', nonstring='simplerepr')
```

Comment:
The changes to this file are being done in the wrong place. Value needs to be converted to a text string when it enters ansible, not at the filter plugin level.

Reply:
Sorry, I should have probably added a comment about why this is being done. It actually isn't necessary for this PR, but is useful.  In upstream jinja2, every value that is passed to a filter is stringified before use.  This ensures that the methods on `StrictUndefined` fire causing an `UndefinedError`, as opposed to something like `TypeError: expected string or bytes-like object`.